### PR TITLE
View annotation: view draw performance

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes

Reduce allocating new collections when drawing views as this is called generally 60 times per second.

### CODE REVIEW

File: sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt [hunk 1]
```diff
@@ -167,12 +169,27 @@ internal class ViewAnnotationManagerImpl(
...
+    idsToAddSet.clear()
+    idsToRemoveSet.clear()
+    positionDescriptorCoreList.forEach {
+      if (!currentViewsDrawnMap.keys.contains(it.identifier)) {
+        idsToAddSet.add(it.identifier)
+      }
+    }
+    currentViewsDrawnMap.keys.forEach { id ->
+      var contains = false
+      for (descriptor in positionDescriptorCoreList) {
```
Comment (alexander-kulikovskii) [inline sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt:181]: how about converting `positionDescriptorCoreList` to set and check `descriptor in positionDescriptorCoreSet`
how large this lists can be? 20, 100, 1000 items?

Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt:181]: Discussed offline, let's go with current solution.

**Review by alexander-kulikovskii — APPROVED** [review]

File: sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt [hunk 2]
```diff
@@ -167,12 +169,27 @@ internal class ViewAnnotationManagerImpl(
...
+        if (descriptor.identifier == id) {
+          contains = true
+          break
+        }
```
Comment (yunikkk) [inline sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt:185]: nit : can be simplified to 
```
if (positionDescriptorCoreList.indexOfFirst { it.identifier == id } == -1) {
  idsToRemoveSet.add(id)
}
```

File: sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt [hunk 3]
```diff
@@ -167,12 +169,27 @@ internal class ViewAnnotationManagerImpl(
...
+      }
+      if (!contains) {
+        idsToRemoveSet.add(id)
+      }
+    }
     // firstly delete views that do not belong to the viewport
-    idsToDeleteSet.forEach {
+    idsToRemoveSet.forEach {
```
Comment (yunikkk) [inline sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt:185]: It's iterated right after loop adding to it, maybe just call this there?

File: sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt (hunk 2 above)
Comment (kiryldz) [reply sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt]: nice

**Review by yunikkk — APPROVED** [review]

**Pull request merged**